packages/react-provider/src/components/FluentProvider/useFluentProvider.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react';
+import { makeMergeProps, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
+import { FluentProviderProps, FluentProviderState } from './FluentProvider.types';
+import { useFluent } from '@fluentui/react-shared-contexts';
+
+export const fluentProviderShorthandProps: (keyof FluentProviderProps)[] = [];
+
+const mergeProps = makeMergeProps<FluentProviderState>({ deepMerge: fluentProviderShorthandProps });
+
+/**
+ * Create the state required to render FluentProvider.
+ *
+ * The returned state can be modified with hooks such as useFluentProviderStyles,
+ * before being passed to renderFluentProvider.
+ *
+ * @param props - props from this instance of FluentProvider
+ * @param ref - reference to root HTMLElement of FluentProvider
+ * @param defaultProps - (optional) default prop values provided by the implementing type
+ */
+export const useFluentProvider = (
+  props: FluentProviderProps,
+  ref: React.Ref<HTMLElement>,
+  defaultProps?: FluentProviderProps,
+): FluentProviderState => {
+  const state = mergeProps(
+    {
+      ref: useMergedRefs(ref, React.useRef(null)),
+      as: 'div',
+    },
+    defaultProps,
+    resolveShorthandProps(props, fluentProviderShorthandProps),
+  );
+
+  const parentContext = useFluent();
+  state.document = state.document ?? parentContext.document;
```

Comment:
```suggestion
  // TODO: add merge functions to "dir" merge, nesting providers with the same "dir" should not add additional attributes to DOM
  // see https://github.com/microsoft/fluentui/blob/0dc74a19f3aa5a058224c20505016fbdb84db172/packages/fluentui/react-northstar/src/utils/mergeProviderContexts.ts#L89-L93
  state.document = state.document ?? parentContext.document;
```

Please restore a TODO